FEXCore/Source/Interface/Core/JIT/Arm64/MiscOps.cpp

```diff
@@ -120,6 +120,37 @@ DEF_OP(SetRoundingMode) {
   msr(ARMEmitter::SystemRegister::FPCR, TMP1);
 }
 
+DEF_OP(PushRoundingMode) {
+  auto Op = IROp->C<IR::IROp_PushRoundingMode>();
+  auto Dest = GetReg(Node);
+
+  // Save the old rounding mode
+  mrs(Dest, ARMEmitter::SystemRegister::FPCR);
+
+  // vixl simulator doesn't support anything beyond ties-to-even rounding
+  if (!CTX->Config.DisableVixlIndirectCalls) [[unlikely]] {
+    return;
+  }
+
+  // Insert the rounding flags, reversing the mode bits as above
+  if (Op->RoundMode == 3) {
+    orr(ARMEmitter::Size::i64Bit, TMP1, Dest, 3 << 22);
+  } else if (Op->RoundMode == 0) {
+    and_(ARMEmitter::Size::i64Bit, TMP1, Dest, ~(3 << 22));
+  } else {
+    and_(ARMEmitter::Size::i64Bit, TMP1, Dest, ~(Op->RoundMode << 22));
```

Comment:
Perhaps have an assert here ensuring the rounding mode is either 1 or 2, so it prevents this from carrying on in the event of invalid values being passed in (either through refactoring in the future, or using this with other things in the future)

Reply:
Done. Good idea, thanks.

Reply:
Could also use the EmitValidation section in IR.json to ensure `RoundMode >= 0 && RoundMode <= 3`